# Setup frames, topics and other generall stuff:
outputTransforms: true # whether or not we want to publish the /tf transforms. Default: true
initFrame: odom   # the initialization of the lidar frame. This is where the
                  # lidar starts, can also be used as our odom frame. Default: /camera_init
mapFrame: loam_map    # frame for the map generated by loam - estimated incrementally. Default: /aft_mapped
odomFrame: loam_odom  # some odom frame that loam creates, DO NOT USE. Default: /laser_odom
lidarFrame: hvlp_link # this is the frame of the lidar estimated continuously. Default: /camera

mapOdomTopic: map_odom   # topic associated with mapFrame. Default: /aft_mapped_to_init
loamOdomTopic: loam_odom # topic associated with odomFrame.Default: /laser_odom_to_init
lidarOdomTopic: lidar_odom # topic associated with lidarFrame. Default: /integrated_to_init

imuInputTopic: /imu/data # Default: /imu/data. Change this to another topic name if you don't want to use IMU.
transformImuData: true # Default: false. If true, it will lookup transform and transform IMU data to lidar frame
imuFrame: IMU1_link  # Default: /imu. Used to lookup transform from imu to lidar.
pointCloudInputTopic: /hvlp/velodyne_points
lidarOdomCov: [0.001, 0.001, 0.001, 0.001, 0.001, 0.001] # Set the covariance used for the lidar odometry topic
              # NOTE: this is the diagonals which represent [Sxx, Syy, Szz, Srxrx, Sryry, Srzrx]

# The following are used in LaserOdometry, LaserMapping, ScanRegistration:
scanPeriod: 0.1 # expected > 0, default 0.1. Time between scans to process

# LaserMapping
maxIterationsMapping: 10 # expected int > 0, default 10. Maximum number of registration iterations
cornerFilterSize: 0.2 # expected >= 0.001, default 0.02 (Kaarta 0.2). Voxel Grid size in all directions for corner clouds
surfaceFilterSize: 0.4 # expected >= 0.001, default 0.4 (Kaarta 0.4). Voxel Grid size in all directions for surface clouds
mapFilterSize: 0.02 # expected >= 0.001, default 0.02. Voxel Grid size in all directions for aggregate map
                    # NOTE: This doesn't seem to be implemented
deltaTAbortMapping: 0.05 # expected > 0, default 0.05. Optimization abort threshold for deltaT (translation)
deltaRAbortMapping: 0.05 # expected > 0, default 0.05. Optimization abort threshold for deltaR (rotation)

# LaserOdometry
ioRatio: 2 # Expected int >= 1, Default 2. Ratio of input to output frames
deltaTAbortOdom: 0.1 # expected > 0, default 0.1. Optimization abort threshold for deltaT (translation)
deltaRAbortOdom: 0.1 # expected > 0, default 0.1. Optimization abort threshold for deltaR (rotation)
maxIterationsOdom: 25 # expected int > 0, default 25. Maximum number of registration iterations

# ScanRegistration
imuHistorySize: 200 # Expected int >= 1, default: 200. The size of the IMU history state buffer.
featureRegions: 6 # Expected int >=1, default: 6. The number of (equally sized) regions used to distribute the feature extraction within a scan
curvatureRegion: 5 # Expected int >=1, default: 5. The number of surrounding points (+/- region around a point) used to calculate a point curvature
maxCornerSharp: 2 # Expected int >=1, default: 5. The maximum number of sharp corner points per feature region.
maxCornerLessSharp: 20 # Expected int >= maxCornerSharp. Default: 10 times maxCornerSharp. The maximum number of less sharp corner points per feature region.
maxSurfaceFlat: 4 # Expected int >=1, default: 4. The maximum number of flat surface points per feature region.
surfaceCurvatureThreshold: 0.1 # Expected >= 0.001, default: 0.1. The curvature threshold below / above a point is considered a flat / corner point.
lessFlatFilterSize: 0.2 # Expected >= 0.001, default: 0.2. The voxel size used for down sizing the remaining less flat surface points.

# TransformMainenance
